File: trunk/MS.NET/clrzbus/examples/pubsub/Subscriber.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus;
namespace zbus
{
    class PubSubWorker
    {
        public static void Main(string[] args)
        {
            ConnectionConfig connCfg = new ConnectionConfig();
            connCfg.Host = "127.0.0.1";
            connCfg.Port = 15555;
            Connection conn = new Connection(connCfg);

            WorkerConfig workerCfg = new WorkerConfig();
            workerCfg.Service = "pubsub";
            workerCfg.Mode = WorkerConfig.MODE_BC; //broadcast, now as PubSub

            Worker worker = new Worker(conn, workerCfg);
            while (true)
            {
                try
                {
                    ZMsg msg = worker.Recv();
                    msg.Dump(); 
                }
                catch (Exception e)
                {
                    Console.WriteLine(e);
                    break;
                }
            }
            conn.Destroy();
            worker.Destroy();
        } 
    }
}

File: JAVA/jzbus/src/test/java/net/zbus/pubsub/Subscriber.java
package net.zbus.pubsub;

import net.zbus.SafeConnection;
import net.zbus.ConnectionConfig;
import net.zbus.Worker;
import net.zbus.WorkerConfig;
import net.zbus.ZMsg;

public class Subscriber {
 
	public static void main(String[] args) { 
		//1) 创建连接到ZBUS总线
		ConnectionConfig connCfg = new ConnectionConfig();
		SafeConnection connection = new SafeConnection(connCfg);
	
		//2) 注册发布订阅服务（MyPubSub）
		WorkerConfig workerCfg = new WorkerConfig();
		workerCfg.setService("MyPubSub");
		workerCfg.setMode(WorkerConfig.MODE_PUBSUB);
		Worker worker = new Worker(connection, workerCfg);
		
		//3) 订阅主题，可多主题订阅
		String topic = "topic1";
		worker.subscribe(topic);
		
		System.out.format("Subscriber on '%s', waiting for message...\n", topic);
		while(true){
			try{
				//等待主题消息
				ZMsg msg = worker.recv();
				
				msg.dump();
			}catch (Exception e) {
				e.printStackTrace();
				break;
			}
		}
		
		//4) 销毁连接相关
		worker.destroy();
		connection.destroy();
	}

}
